### ISSUE TITLE
returns NaN for free and size, and '' for diskPath when powershell settings have been adjusted on the user's machine

### ISSUE THREAD

**Issue by davidburson** [Alex-D/check-disk-space#23]
My powershell settings cause blank lines to be included `parsed` in `mapOutput`, and `filter` does not remove them, so `filtered[0]` ends up being an empty string, which causes the invalid return value.

I'll post a PR shortly.

**Comment by davidburson** [comment]
I don't have permission to push a new branch, so here's the diff that shows the fix.  I just added `driveLetter && ` to the beginning of the return.
```
diff --git a/src/index.ts b/src/index.ts
index 27aafee..07a7e42 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -129,7 +129,9 @@ function checkDiskSpace(directoryPath: string, dependencies: Dependencies = {
                        driveData => {
                                // Only get the drive which match the path
                                const driveLetter = driveData[0]
-                               return directoryPath.toUpperCase().startsWith(driveLetter.toUpperCase())
+                               // driveData[0] can be empty string if the user has modified settings related to powershell,
+                               //      causing true to be wrongly returned unless we explicitly check for that case
+                               return driveLetter && directoryPath.toUpperCase().startsWith(driveLetter.toUpperCase())
                        },
                        {
                                diskPath: 0,
```

**Comment by johnheroy** [comment]
@davidburson what powershell settings did you change? curious how to reproduce this

**Comment by Nicky03** [comment]
> 



> I don't have permission to push a new branch, so here's the diff that shows the fix. I just added `driveLetter && ` to the beginning of the return.
> 
> ```diff
> diff --git a/src/index.ts b/src/index.ts
> index 27aafee..07a7e42 100644
> --- a/src/index.ts
> +++ b/src/index.ts
> @@ -129,7 +129,9 @@ function checkDiskSpace(directoryPath: string, dependencies: Dependencies = {
>                         driveData => {
>                                 // Only get the drive which match the path
>                                 const driveLetter = driveData[0]
> -                               return directoryPath.toUpperCase().startsWith(driveLetter.toUpperCase())
> +                               // driveData[0] can be empty string if the user has modified settings related to powershell,
> +                               //      causing true to be wrongly returned unless we explicitly check for that case
> +                               return driveLetter && directoryPath.toUpperCase().startsWith(driveLetter.toUpperCase())
>                         },
>                         {
>                                 diskPath: 0,
> ```

@davidburson Now I also encounter the situation of returning 'NAN'. I would like to ask, after adding 'driveLetter &&', can I get the correct value?

**Comment by Alex-D** [comment]
This will be fixed in the next release :)

I've added a filter to ignore empty lines, should do the work.

**Comment by Alex-D** [comment]
Now fixed in 3.4.0

**Comment by davidburson** [comment]
> @davidburson what powershell settings did you change? curious how to reproduce this

Sorry for the very slow reply.  I can't remember for sure, but here's the changed settings I found, in case it's helpful:
```
# From https://dev.to/ofhouse/add-a-bash-like-autocomplete-to-your-powershell-4257
# Shows navigable menu of all options when hitting Tab
Set-PSReadlineKeyHandler -Key Tab -Function MenuComplete

# Autocompletion for arrow keys
Set-PSReadlineKeyHandler -Key UpArrow -Function HistorySearchBackward
Set-PSReadlineKeyHandler -Key DownArrow -Function HistorySearchForward
```
